aquarium/src/cg/group4/view/AquariumScreen.java

```diff
@@ -0,0 +1,249 @@
+package cg.group4.view;
+
+import cg.group4.data_structures.Pair;
+import cg.group4.data_structures.collection.Collection;
+import cg.group4.data_structures.collection.collectibles.Collectible;
+import cg.group4.view.screen_mechanics.GameSkin;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.utils.Align;
+
+import java.util.HashSet;
+import java.util.Observable;
+import java.util.Observer;
+
+/**
+ * Screen which displays the collected fish in an aquarium.
+ */
+public class AquariumScreen implements Screen {
+
+    /**
+     * Stage for the content.
+     */
+    protected final Stage cStage;
+
+    /**
+     * Set containing all displayed elements.
+     */
+    protected HashSet<CollectibleRenderer> cDisplayRendererSet,
+        cInitialSet;
+
+    protected Collection cDisplaySet;
+
+    /**
+     * Style for layout items.
+     */
+    protected final GameSkin cStyle;
+
+    /**
+     * Label to display owner information after having clicked on a collectible.
+     */
+    protected Label cOwnerLabel;
+
+    /**
+     * Label to display catch date information after having clicked on a collectible.
+     */
+    protected Label cDateLabel;
+
+    /**
+     * Label to display catch date information after having clicked on a collectible.
+     */
+    protected Label cStatusLabel;
+
+
+    /**
+     * Table to hold layout items.
+     */
+    protected Table cLabelTable;
+
+    /**
+     * Observer for the collection.
+     */
+    protected Observer cCollectionObserver;
+
+    /**
+     * Observer for the info label.
+     */
+    protected Observer cLabelObserver;
+
+    /**
+     * Initializes the aquarium.
+     */
+    public AquariumScreen() {
+
+        createCollectionObserver();
+        createLabelObserver();
+
+        cInitialSet = new HashSet<CollectibleRenderer>();
+        cDisplayRendererSet = new HashSet<CollectibleRenderer>();
+
+        cDisplaySet = new Collection("$$Nil");
+
+        cStyle = new GameSkin();
+        cStyle.createUIElements(720);
+        cStage = new Stage();
+
+        initStage();
+        initTooltipLabels();
+    }
+
+    /**
+     * Adds an observer for the collection update, which is received from the server.
+     */
+    public void createCollectionObserver() {
+        cCollectionObserver = new Observer() {
+            @Override
+            public void update(Observable o, Object arg) {
+                Collection collection = (Collection) arg;
+                Gdx.app.log(this.getClass().getName(), "Received Collection update: " + collection.toString());
+
+
+                if (collection.isEmpty()) {
+                    notifyUserEmptyContainerReceived();
+                }
+
+                for (Collectible c : collection) {
```

Comment:
these 2 for loops can be placed in the else case of the collection.isEmpty() check, I assume?